Mathlib/GroupTheory/MonoidLocalization.lean

```diff
@@ -1866,3 +1888,158 @@ end LocalizationWithZeroMap
 end Submonoid
 
 end CommMonoidWithZero
+
+namespace Localization
+
+variable {α : Type _} [CancelCommMonoid α] {s : Submonoid α} {a₁ b₁ : α} {a₂ b₂ : s}
+
+@[to_additive]
+theorem mk_left_injective (b : s) : Injective fun a => mk a b := fun c d h => by
+  have : Nonempty s := ⟨b⟩ -- porting note: Needed to add this `have`
```

Comment:
Do you know why? Did we misport a lemma and add this hypothesis unecessarily?

Reply:
No, I cannot find the lemma that uses it. It might be a matter of instance unification, *again*.

Reply:
I think we're missing the fact that submonoids are inhabited (with 1)

Reply:
In Lean3 this finds `has_one.nonempty`

Reply:
No fix in sight, then?

Reply:
I think TC search just times out before it finds it